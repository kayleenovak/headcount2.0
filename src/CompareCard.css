.compare-card {
  border-radius: 10px;
  box-shadow: 2px 2px 5px black;
  padding: 10px;
  background: white;
  height: 300px;
  width: 250px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  border: 4px solid grey;
  margin-bottom: 20px;
}