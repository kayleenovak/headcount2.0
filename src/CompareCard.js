import React from 'react';
import './CompareCard.css'

const CompareCard = (props) => {
  console.log('Compare card props',  props)
  const stats = Object.keys(props.schoolStats).map(stat => {
    return <li className={props.schoolStats[stat] >= 1 ? 'greater-than' : 'less-than'}>{ stat }: { props.schoolStats[stat] }</li>
  })
  return (
    <div className='compare-card' onClick={ props.removeCompare }>
      <h3>{ props.school }</h3>
      <button onClick={ props.removeComparedSchool } id={ props.school }>Delete</button>
      { stats }
    </div>
    )
}

export default CompareCard