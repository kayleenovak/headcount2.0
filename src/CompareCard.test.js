import React from 'react';
import { shallow, mount } from 'enzyme';
import CompareCard from './CompareCard.js'

describe('Compare Card', () => {
  it('should match the snapshot', () => {
    const mockFunction = jest.fn()
    const mockSchools = ['COLORADO']
    const mockStats = {
      '2004': 1,
      '2005': 1
    }
    const wrapper = shallow(<CompareCard removeComparedSchool={ mockFunction } school={ mockSchools[0] } schoolStats={ mockStats } />)

    expect(wrapper).toMatchSnapshot()
  })
})