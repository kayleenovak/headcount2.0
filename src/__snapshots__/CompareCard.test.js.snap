// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compare Card should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CompareCard
    removeComparedSchool={[Function]}
    school="COLORADO"
    schoolStats={
        Object {
            "2004": 1,
            "2005": 1,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          COLORADO
</h3>,
        <button
          id="COLORADO"
          onClick={[Function]}
>
          Delete
</button>,
        Array [
          <li>
            2004
            : 
            1
</li>,
          <li>
            2005
            : 
            1
</li>,
        ],
      ],
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "COLORADO",
        },
        "ref": null,
        "rendered": "COLORADO",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Delete",
          "id": "COLORADO",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Delete",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "2004",
            ": ",
            1,
          ],
        },
        "ref": null,
        "rendered": Array [
          "2004",
          ": ",
          1,
        ],
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "2005",
            ": ",
            1,
          ],
        },
        "ref": null,
        "rendered": Array [
          "2005",
          ": ",
          1,
        ],
        "type": "li",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            COLORADO
</h3>,
          <button
            id="COLORADO"
            onClick={[Function]}
>
            Delete
</button>,
          Array [
            <li>
              2004
              : 
              1
</li>,
            <li>
              2005
              : 
              1
</li>,
          ],
        ],
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "COLORADO",
          },
          "ref": null,
          "rendered": "COLORADO",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Delete",
            "id": "COLORADO",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Delete",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "2004",
              ": ",
              1,
            ],
          },
          "ref": null,
          "rendered": Array [
            "2004",
            ": ",
            1,
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "2005",
              ": ",
              1,
            ],
          },
          "ref": null,
          "rendered": Array [
            "2005",
            ": ",
            1,
          ],
          "type": "li",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
